pilot/pkg/networking/grpcgen/lds.go

```diff
@@ -281,6 +285,28 @@ func buildOutboundListeners(node *model.Proxy, push *model.PushContext, filter l
 				if !match.includesPort(sPort) {
 					continue
 				}
+				filters := supportedFilters
+				sessionCookie := sv.Attributes.Labels["x-session-cookie"]
+				if sessionCookie != "" {
+					filters = append(filters, &hcm.HttpFilter{
+						Name: "envoy.filters.http.stateful_session", // TODO: wellknown.
+						ConfigType: &hcm.HttpFilter_TypedConfig{
+							TypedConfig: protoconv.MessageToAny(&stateful_sessionv3.StatefulSession{
+								SessionState: &core.TypedExtensionConfig{
+									Name: "envoy.http.stateful_session.cookie",
+									TypedConfig: protoconv.MessageToAny(&cookiev3.CookieBasedSessionState{
+										Cookie: &httpv3.Cookie{
+											Path: "/",
+											Ttl:  &durationpb.Duration{Seconds: 120},
```

Comment:
So these (`Path` and `Ttl`) are fixed and not configurable from the control plane API? For a temporary solution this might be okay but wondering if it's easy to get the value from attrs similar to the cookie name (that we get from `sv.Attributes.Labels["x-session-cookie"]`)

Reply:
TTL - we can make it configurable. 

The Path is VERY complicated - sticky sessions are also very common with end users ( browsers and gateway ), and even in the mesh one destination (service) may be the backend for multiple routes. 

If we were attaching this to a HTTPRoute or GRPCRoute - the path would be automatically populated from the route matching rule, we wouldn't want to allow user to specify path "/foo" if the service is routed as "/bar". 

I would defer the Path for a future change.

Reply:
What would be a reasonable default ? 1 h ? And should I add an annotation as well, or leave it to the final API ? 

120s is a bit too short.

session-cookie-max-age ( mirroring nginx annotation ) ? I see traefik doesn't have a setting for this, tomcat session timeout default was 30 min.

Reply:
> What would be a reasonable default ? 1 h ?

Sounds good to me

>  And should I add an annotation as well, or leave it to the final API ?

Annotation will be good.

> 
> 120s is a bit too short.
> 
> session-cookie-max-age ( mirroring nginx annotation ) ?

If that's the name/annotation you are suggesting, sounds good to me.